plugin/slime.vim

```diff
@@ -52,22 +52,41 @@ endfunction
 " Tmux
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
+function! s:TmuxDefaultSocket()
+  " The socket path is the first value in the comma-separated list of $TMUX.
+  return empty($TMUX) ? "default" : split($TMUX, ",")[0]
+endfunction
+
+function! s:IsAbsolute(path)
+  " Case sensitivity does not matter here, but let's follow good practice.
+  " TODO: Make this cross-platform. Windows supports tmux as of mid-2016.
+  return a:path[0] ==? "/"
+endfunction
+
+function! s:TmuxCommand(config, args)
+  let l:socket = a:config["socket_name"]
+  " Use tmux -S for an absolute path to the socket.
+  " Use tmux -L for a relative path to the socket in tmux's temporary directory.
+  let l:socket_option = s:IsAbsolute(l:socket) ? "-S" : "-L"
+  return system("tmux " . l:socket_option . " " . shellescape(l:socket) . " " . a:args)
+endfunction
+
 function! s:TmuxSend(config, text)
   call s:WritePasteFile(a:text)
-  call system("tmux -L " . shellescape(a:config["socket_name"]) . " load-buffer " . g:slime_paste_file)
-  call system("tmux -L " . shellescape(a:config["socket_name"]) . " paste-buffer -d -t " . shellescape(a:config["target_pane"]))
+  call s:TmuxCommand(a:config, "load-buffer " . g:slime_paste_file)
+  call s:TmuxCommand(a:config, "paste-buffer -d -t " . shellescape(a:config["target_pane"]))
 endfunction
 
 function! s:TmuxPaneNames(A,L,P)
   let format = '#{pane_id} #{session_name}:#{window_index}.#{pane_index} #{window_name}#{?window_active, (active),}'
-  return system("tmux -L " . shellescape(b:slime_config['socket_name']) . " list-panes -a -F " . shellescape(format))
+  return s:TmuxCommand(b:slime_config, "list-panes -a -F " . shellescape(format))
 endfunction
 
 function! s:TmuxConfig() abort
   if !exists("b:slime_config")
-    let b:slime_config = {"socket_name": "default", "target_pane": ":"}
+    let b:slime_config = {"socket_name": s:TmuxDefaultSocket(), "target_pane": ":.2"}
```

Comment:
To explain `:.2`, I generally run vim in a tmux window, and then split the window into two panes when I want to start a REPL. The target `:.2` sends text to the other pane in the same window, to the REPL. The default target `:` sends the text to the pane running vim, and if I'm in normal mode, God help me.

I'll move this to the README too.